diff -ruN xf86-video-amdgpu-23.0-2/src/amdgpu_kms.c xf86-video-amdgpu-23.0.0/src/amdgpu_kms.c
--- xf86-video-amdgpu-23.0-2/src/amdgpu_kms.c	2023-02-22 12:14:50.000000000 -0500
+++ xf86-video-amdgpu-23.0.0/src/amdgpu_kms.c	2024-11-09 23:09:08.343371082 -0500
@@ -1510,14 +1510,14 @@
 	else if (info->family < AMDGPU_FAMILY_CI)
 		info->cursor_w = CURSOR_WIDTH;
 	else
-		info->cursor_w = CURSOR_WIDTH_CIK;
+		info->cursor_w = CURSOR_WIDTH;
 
 	if (drmGetCap(fd, DRM_CAP_CURSOR_HEIGHT, &value) == 0)
 		info->cursor_h = value;
 	else if (info->family < AMDGPU_FAMILY_CI)
 		info->cursor_h = CURSOR_HEIGHT;
 	else
-		info->cursor_h = CURSOR_HEIGHT_CIK;
+		info->cursor_h = CURSOR_HEIGHT;
 }
 
 /* When the root window is mapped, set the initial modes */
