diff -ruN src/data/amdgpu.ids src2/data/amdgpu.ids
--- src/data/amdgpu.ids	2024-08-26 16:10:22.000000000 -0400
+++ src2/data/amdgpu.ids	2024-11-09 22:18:30.972862601 -0500
@@ -584,3 +584,7 @@
 98E4,	EA,	AMD Radeon R4 Graphics
 98E4,	EB,	AMD Radeon R3 Graphics
 98E4,	EB,	AMD Radeon R4 Graphics
+9920,	00, AMD Liverpool (PlayStation 4)
+9922,	00, AMD Liverpool (PlayStation 4)
+9923,	00, AMD Liverpool (PlayStation 4)
+9924,	11, AMD Gladius (PlayStation 4)
\ No newline at end of file
diff -ruN src/radeon/r600_pci_ids.h src2/radeon/r600_pci_ids.h
--- src/radeon/r600_pci_ids.h	2024-08-26 16:10:22.000000000 -0400
+++ src2/radeon/r600_pci_ids.h	2024-11-09 22:19:02.688896446 -0500
@@ -473,6 +473,11 @@
 CHIPSET(0x131C, KAVERI_131C, KAVERI)
 CHIPSET(0x131D, KAVERI_131D, KAVERI)
 
+CHIPSET(0x9920, LIVERPOOL_9920, LIVERPOOL)
+CHIPSET(0x9922, LIVERPOOL_9922, LIVERPOOL)
+CHIPSET(0x9923, LIVERPOOL_9923, LIVERPOOL)
+CHIPSET(0x9924, GLADIUS_9924, GLADIUS)
+
 CHIPSET(0x67A0, HAWAII_67A0, HAWAII)
 CHIPSET(0x67A1, HAWAII_67A1, HAWAII)
 CHIPSET(0x67A2, HAWAII_67A2, HAWAII)
diff -ruN src/radeon/radeon_surface.c src2/radeon/radeon_surface.c
--- src/radeon/radeon_surface.c	2024-08-26 16:10:22.000000000 -0400
+++ src2/radeon/radeon_surface.c	2024-11-09 22:19:21.688917554 -0500
@@ -86,6 +86,8 @@
     CHIP_HAINAN,
     CHIP_BONAIRE,
     CHIP_KAVERI,
+    CHIP_LIVERPOOL,
+    CHIP_GLADIUS,
     CHIP_KABINI,
     CHIP_HAWAII,
     CHIP_MULLINS,
