diff -ruN a/include/pci_ids/radeonsi_pci_ids.h b/include/pci_ids/radeonsi_pci_ids.h
--- a/include/pci_ids/radeonsi_pci_ids.h	2024-10-30 14:12:27.000000000 -0400
+++ b/include/pci_ids/radeonsi_pci_ids.h	2024-11-09 22:41:37.270124283 -0500
@@ -88,6 +88,11 @@
 CHIPSET(0x665D, BONAIRE)
 CHIPSET(0x665F, BONAIRE)
 
+CHIPSET(0x9920, LIVERPOOL)
+CHIPSET(0x9922, LIVERPOOL)
+CHIPSET(0x9923, LIVERPOOL)
+CHIPSET(0x9924, GLADIUS)
+
 CHIPSET(0x9830, KABINI)
 CHIPSET(0x9831, KABINI)
 CHIPSET(0x9832, KABINI)
diff -ruN a/src/amd/addrlib/src/amdgpu_asic_addr.h b/src/amd/addrlib/src/amdgpu_asic_addr.h
--- a/src/amd/addrlib/src/amdgpu_asic_addr.h	2024-10-30 14:12:27.000000000 -0400
+++ b/src/amd/addrlib/src/amdgpu_asic_addr.h	2024-11-09 22:42:43.018094832 -0500
@@ -62,7 +62,9 @@
 #define AMDGPU_HAWAII_RANGE     0x28, 0x3C //# 40 <= x < 60
 
 #define AMDGPU_SPECTRE_RANGE    0x01, 0x41 //#   1 <= x < 65
-#define AMDGPU_SPOOKY_RANGE     0x41, 0x81 //#  65 <= x < 129
+#define AMDGPU_SPOOKY_RANGE     0x41, 0x61
+#define AMDGPU_STARSHA_RANGE    0x61, 0x71
+#define AMDGPU_STARSHP_RANGE    0x71, 0x81
 #define AMDGPU_KALINDI_RANGE    0x81, 0xA1 //# 129 <= x < 161
 #define AMDGPU_GODAVARI_RANGE   0xA1, 0xFF //# 161 <= x < max
 
@@ -134,6 +136,8 @@
 
 #define ASICREV_IS_SPECTRE(r)          ASICREV_IS(r, SPECTRE)
 #define ASICREV_IS_SPOOKY(r)           ASICREV_IS(r, SPOOKY)
+#define ASICREV_IS_STARSHA(r)           ASICREV_IS(r, STARSHA)
+#define ASICREV_IS_STARSHP(r)           ASICREV_IS(r, STARSHP)
 #define ASICREV_IS_KALINDI(r)          ASICREV_IS(r, KALINDI)
 #define ASICREV_IS_KALINDI_GODAVARI(r) ASICREV_IS(r, GODAVARI)
 
diff -ruN a/src/amd/common/ac_gpu_info.c b/src/amd/common/ac_gpu_info.c
--- a/src/amd/common/ac_gpu_info.c	2024-10-30 14:12:27.000000000 -0400
+++ b/src/amd/common/ac_gpu_info.c	2024-11-09 22:49:25.938163846 -0500
@@ -837,6 +837,8 @@
       case FAMILY_KV:
          identify_chip2(SPECTRE, KAVERI);
          identify_chip2(SPOOKY, KAVERI);
+         identify_chip2(STARSHA, LIVERPOOL);
+         identify_chip2(STARSHP, GLADIUS);
          identify_chip2(KALINDI, KABINI);
          identify_chip2(GODAVARI, KABINI);
          break;
@@ -1166,6 +1168,8 @@
       case CHIP_VERDE:
       case CHIP_HAINAN:
       case CHIP_BONAIRE:
+      case CHIP_LIVERPOOL:
+      case CHIP_GLADIUS:
       case CHIP_KAVERI:
       case CHIP_ICELAND:
       case CHIP_CARRIZO:
@@ -2127,6 +2131,8 @@
    case CHIP_PITCAIRN:
    case CHIP_VERDE:
    case CHIP_BONAIRE:
+   case CHIP_LIVERPOOL:
+   case CHIP_GLADIUS:
    case CHIP_HAWAII:
    case CHIP_TONGA:
    case CHIP_FIJI:
@@ -2177,6 +2183,14 @@
       raster_config = 0x16000012;
       raster_config_1 = 0x00000000;
       break;
+   case CHIP_LIVERPOOL:
+      raster_config = 0x2a00161a;
+      raster_config_1 = 0x00000000;
+      break;
+   case CHIP_GLADIUS:
+      raster_config = 0x2a00161a;
+      raster_config_1 = 0x0000002e;
+      break;
    /* 2 SEs / 8 RBs */
    case CHIP_TAHITI:
    case CHIP_PITCAIRN:
diff -ruN a/src/amd/common/amd_family.c b/src/amd/common/amd_family.c
--- a/src/amd/common/amd_family.c	2024-10-30 14:12:27.000000000 -0400
+++ b/src/amd/common/amd_family.c	2024-11-09 22:47:04.026100236 -0500
@@ -23,6 +23,8 @@
    CASE(OLAND);
    CASE(HAINAN);
    CASE(BONAIRE);
+   CASE(LIVERPOOL);
+   CASE(GLADIUS);
    CASE(KABINI);
    CASE(KAVERI);
    CASE(HAWAII);
@@ -108,6 +110,10 @@
       return "hainan";
    case CHIP_BONAIRE:
       return "bonaire";
+   case CHIP_LIVERPOOL:
+      return "bonaire";
+   case CHIP_GLADIUS:
+      return "bonaire";
    case CHIP_KABINI:
       return "kabini";
    case CHIP_KAVERI:
diff -ruN a/src/amd/common/amd_family.h b/src/amd/common/amd_family.h
--- a/src/amd/common/amd_family.h	2024-10-30 14:12:27.000000000 -0400
+++ b/src/amd/common/amd_family.h	2024-11-09 22:45:05.646076728 -0500
@@ -80,6 +80,8 @@
    CHIP_HAINAN,
    /* GFX7 (Sea Islands) */
    CHIP_BONAIRE,
+   CHIP_LIVERPOOL,
+   CHIP_GLADIUS,
    CHIP_KAVERI,
    CHIP_KABINI,
    CHIP_HAWAII,         /* Radeon 290, 390 */
diff -ruN a/src/amd/vulkan/winsys/null/radv_null_winsys.c b/src/amd/vulkan/winsys/null/radv_null_winsys.c
--- a/src/amd/vulkan/winsys/null/radv_null_winsys.c	2024-10-30 14:12:27.000000000 -0400
+++ b/src/amd/vulkan/winsys/null/radv_null_winsys.c	2024-11-09 22:47:16.150104289 -0500
@@ -27,6 +27,10 @@
    [CHIP_OLAND] = {0x6060, 2, true},
    [CHIP_HAINAN] = {0x6660, 2, true},
    [CHIP_BONAIRE] = {0x6640, 4, true},
+   [CHIP_LIVERPOOL] = { 0x9920, 2, false },
+   [CHIP_LIVERPOOL] = { 0x9922, 2, false },
+   [CHIP_LIVERPOOL] = { 0x9923, 2, false },
+   [CHIP_GLADIUS] = { 0x9924, 4, false },
    [CHIP_KAVERI] = {0x1304, 2, false},
    [CHIP_KABINI] = {0x9830, 2, false},
    [CHIP_HAWAII] = {0x67A0, 16, true},
diff -ruN a/src/gallium/targets/d3dadapter9/description.c b/src/gallium/targets/d3dadapter9/description.c
--- a/src/gallium/targets/d3dadapter9/description.c	2024-10-30 14:12:27.000000000 -0400
+++ b/src/gallium/targets/d3dadapter9/description.c	2024-11-09 22:47:34.306110876 -0500
@@ -36,6 +36,8 @@
     {"KAVERI",                      "AMD Radeon(TM) R7 Graphics"},
     {"KABINI",                      "AMD Radeon HD 8400 / R3 Series"},
     {"BONAIRE",                     "AMD Radeon HD 8770"},
+    {"LIVERPOOL",                   "AMD Liverpool (Playstation 4)"},
+    {"GLADIUS",                     "AMD Gladius (Playstation 4)"},
     {"OLAND",                       "AMD Radeon HD 8670"},
     {"HAINAN",                      "AMD Radeon HD 8600M Series"},
     {"TAHITI",                      "AMD Radeon HD 7900 Series"},
diff -ruN a/src/gallium/winsys/radeon/drm/radeon_drm_winsys.c b/src/gallium/winsys/radeon/drm/radeon_drm_winsys.c
--- a/src/gallium/winsys/radeon/drm/radeon_drm_winsys.c	2024-10-30 14:12:27.000000000 -0400
+++ b/src/gallium/winsys/radeon/drm/radeon_drm_winsys.c	2024-11-09 22:48:41.374140270 -0500
@@ -259,6 +259,8 @@
       ws->info.gfx_level = GFX6;
       break;
    case CHIP_BONAIRE:
+   case CHIP_LIVERPOOL:
+   case CHIP_GLADIUS:
    case CHIP_KAVERI:
    case CHIP_KABINI:
    case CHIP_HAWAII:
@@ -472,6 +474,8 @@
    case CHIP_VERDE:
    case CHIP_OLAND:
    case CHIP_BONAIRE:
+   case CHIP_LIVERPOOL:
+   case CHIP_GLADIUS:
    case CHIP_KAVERI:
       ws->info.max_tcc_blocks = 4;
       break;
@@ -501,9 +505,11 @@
       case CHIP_TAHITI:
       case CHIP_PITCAIRN:
       case CHIP_BONAIRE:
+      case CHIP_LIVERPOOL:
          ws->info.max_se = 2;
          break;
       case CHIP_HAWAII:
+      case CHIP_GLADIUS:
          ws->info.max_se = 4;
          break;
       }
--- a/src/gallium/winsys/radeon/drm/radeon_surface.c   2025-04-05 17:11:35.970953234 -0400
+++ b/src/gallium/winsys/radeon/drm/radeon_surface.c    2025-04-05 17:11:51.679020550 -0400
@@ -78,6 +78,8 @@
     CHIP_OLAND,
     CHIP_HAINAN,
     CHIP_BONAIRE,
+    CHIP_LIVERPOOL,
+    CHIP_GLADIUS,
     CHIP_KAVERI,
     CHIP_KABINI,
     CHIP_HAWAII,
